### ISSUE TITLE
Scientific and Engineering Notations bug out with small numbers

### ISSUE THREAD

**Issue by Kidsheldon** [Aktanusa/CookieMonster#282]
![image](https://user-images.githubusercontent.com/17099837/69904004-261db480-136f-11ea-8918-b396455e4d77.png)
In any field that is set to be formatted, Cookie Monster will attempt to format very small numbers that very much shouldn't be formatted. I suggest that it should only begin formatting numbers starting at 1 million.

**Comment by Kidsheldon** [comment]
On a similar note, Metric does this buggy thing that normally happens with Short Numbers off in the game's setting scale, except it's even more nonsense thanks to the suffix. Not sure what the solution to this would be. Maybe just default to Scientific after a certain point?
![image](https://user-images.githubusercontent.com/17099837/69904048-a512ed00-136f-11ea-874a-b5c1394bac86.png)

**Comment by Aktanusa** [comment]
I'm not sure if you are the same person on reddit who mention the same issue.  If you are, thanks for making the issue!  I plan to make this an option.

**Comment by Kidsheldon** [comment]
Different person, but you're welcome!

**Comment by raekuul** [comment]
I would like to see the one million threshold applied in the micro-direction as well as the macro-direction. This is both silly and hard to parse.
![image](https://user-images.githubusercontent.com/11349476/79279096-414f1380-7e7b-11ea-8a10-c630a810556b.png)

**Comment by DanielNoord** [comment]
I believe #371 has fixed this.
I have tested for a number of cases and did not find any new bugs. I have made the metric-notation revert to scientific notation after it reaches e27.

 @raekuul if you want to confirm that the bug you mentioned has been fixed as well, that would be appreciated!

**Comment by raekuul** [comment]
It does not seem to be fixed, though I may be loading the wrong branch (dev branch and beta branch both give "the game version is too new" warnings)

![1606736640_chrome](https://user-images.githubusercontent.com/11349476/100606924-24673380-32d8-11eb-9d92-e8ed31cc6593.jpg)

**Comment by DanielNoord** [comment]
How are you loading the dev-branch?

**Comment by raekuul** [comment]
per the instructions in README.md, under Bugs and Suggestions

> Before submitting a bug, make sure to give a shot at the latest version of the addon on the dev branch. For this, use the following bookmarklet:

> [(javascript: (function () {
	Game.LoadMod('https://cdn.rawgit.com/Aktanusa/CookieMonster/dev/CookieMonster.js');
}());
)]

**Comment by DanielNoord** [comment]
Hmm, there seems to be something wrong with that link.
@Aktanusa do you know if there is something broken in the uploading procedures of the dev-branch to gh-pages/rawgit?

Sorry about that @raekuul.
In the mean time you can use [this code](https://raw.githubusercontent.com/Aktanusa/CookieMonster/dev/CookieMonster.js) to test the dev-branch.
If you do right mouse button > inspect page > console and then paste all that code in at once you should get the dev-branch of CM running.

**Comment by raekuul** [comment]
runs into Chrome's Cross-Origin Read Blocking (CORB), so I can't test it immediately. Give me a couple hours to change workstations.

**Comment by Aktanusa** [comment]
@DanielNoord Might just be an outdated link.

**Comment by DanielNoord** [comment]
Don't load it via `Game.LoadMod`. Just select all the code in that link and copy-paste directly into the console.
For example, in Safari this looks like this:
<img width="1371" alt="Schermafbeelding 2020-11-30 om 13 00 50" src="https://user-images.githubusercontent.com/13665637/100608234-6611d180-330c-11eb-87db-b4b3867ca7d6.png">

**Comment by DanielNoord** [comment]
@Aktanusa Hmm, I am not sure. For example, in the initial comment to me the sugar lump could is indeed a bit ugly. That shouldn't be formatted (imo). I can think of other scenario's where using e1 or e4 would be a bit ugly.
I think a cut-off of 1 million works well as this is a fairly low number of cookies or cps (which are the areas where formatting is most need), but will catch most instances were formatting might seem out of place.

Obviously, it is very easy to change the cut-off point in the code but personally I agree with it being 1 million (or 999999 to be specific).

**Comment by Aktanusa** [comment]
I can see where the notation can be ugly.  I still think it should just be an option with the cutoff on by default.  There could also be another option to set the cutoff.  I see this as an enhancement.

**Comment by DanielNoord** [comment]
Good idea. I will create a new issue that focus on setting the cutoff point.
For now, I hope that the current fix has resolved the issues of @raekuul and we can use this in the next release.

**Comment by raekuul** [comment]
I am glad to report that the tooltip for Communal Brainsweep is now completely legible under Scientific Notation. Thank you!
